### PR TITLE
Add StopButtonTapped action

### LINKED ISSUE #70: Add StopButtonTapped action
## Description
Action to stop the currently running time entry from the running time entry bottom card.

## Dependencies
#69 

## Definition of done
- [x] The new action is added
- [x] The new action is handled by the reducer and the effect returned
- [x] Tests are added

### PR DESCRIPTION
**Pull request by sofokli1**
### Summary
Adds the StopButtonTapped action and test. also connects the action to the UI

### Relationships
Closes #70

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [x] The code used custom templates

### Pointers
Just check the code and run it once to stop a TE

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
🦑

### CODE REVIEW

File: Modules/Timer/Timer/RunningTimrEntry/RunningTimeEntryReducer.swift [hunk 1]
```diff
@@ -6,7 +6,19 @@ import Repository
...
-    return Reducer { _, _ in
-        return []
+    return Reducer { state, action in
+        switch action {
+        case .stopButtonTapped:
+            guard var runningTimeEntry = state.runningTimeEntry else { return [] }
+            runningTimeEntry.duration = time.now().timeIntervalSince(runningTimeEntry.start)
+            state.entities.timeEntries[runningTimeEntry.id] = runningTimeEntry
+
+            stopTimeEntryEffect(state.entities.timeEntries[runningTimeEntry.id]!, repository: repository)
```
Comment (Odrakir) [inline Modules/Timer/Timer/RunningTimrEntry/RunningTimeEntryReducer.swift:16]: This should be returned, the effect. I know it's empty right now, but in this case just return `Effect.empty`

Also... are we setting the duration here or in the repository... 🤔 @heytherewill ?

Comment (sofokli1) [reply Modules/Timer/Timer/RunningTimrEntry/RunningTimeEntryReducer.swift:16]: I had it as `Effect.empty` but the assertion in the test was complaining

File: Modules/Timer/TimerTests/Reducers/RunningTimeEntryReducerTests.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        let state = RunningTimeEntryState(entities: entities)
+
+        assertReducerFlow(
+            initialState: state,
+            reducer: reducer,
+            steps:
+            Step(.send, .stopButtonTapped) {
+                $0.entities.timeEntries[1]!.duration = 200
+            })
+    }
```
Comment (sofokli1) [inline Modules/Timer/TimerTests/Reducers/RunningTimeEntryReducerTests.swift:37]: @Odrakir If I use an `Effect.empty` in stopTimeEntryEffect, the above test I get hit with this failure https://github.com/toggl/ios/blob/eugene/stopRunningTE/Modules/Timer/TimerTests/TestUtils/ReducerAssert.swift#L83

File: Modules/Timer/Timer/RunningTimrEntry/RunningTimeEntryReducer.swift (hunk 1 above)
Comment (heytherewill) [reply Modules/Timer/Timer/RunningTimrEntry/RunningTimeEntryReducer.swift:16]: @Odrakir We're setting it in the repo, but I guess we should, instead, provide the stop date to the repo

File: Modules/Timer/Timer/Running/Views/RunningTimeEntryViewController.swift [hunk 3]
```diff
@@ -35,6 +35,11 @@ public class RunningTimeEntryViewController: UIViewController, Storyboarded {
             .drive(onNext: updateUI)
             .disposed(by: disposeBag)
 
+        stopButtom.rx.tap
```
Comment (juanlaube) [inline Modules/Timer/Timer/Running/Views/RunningTimeEntryViewController.swift:38]: Just a small typo

```suggestion
        stopButton.rx.tap
```

**Pull request merged**